### PR TITLE
Docs: Add Nginx loadbalancing example with sticky mxid for workers

### PR DESCRIPTION
**Pull request by tswfi**
Add example nginx configuration snippet that

* does load balancing for workers
* respects mxid part of the token
  * from both url parameter and auth header
  * and handles since parameter

Thanks to @olmari for pushing me to write this and testing the configs

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: docs/workers.md [hunk 1]
```diff
@@ -333,6 +333,76 @@ when a user logs in on a new device and can be *very* resource intensive, so
 isolating these requests will stop them from interfering with other users ongoing
 syncs.
 
+Example `nginx` configuration snippet that handles the cases above. This is just an
+example and probably requires some changes according to your particular setup:
+
+```js
```
Comment (dklimpel) [inline docs/workers.md:339]:
```suggestion
```nginx
```

File: docs/workers.md [hunk 2]
```diff
@@ -333,6 +333,76 @@ when a user logs in on a new device and can be *very* resource intensive, so
...
+# Sync initial/normal
+location ~ ^/_matrix/client/(r0|v3)/sync$ {
+	include snippets/matrix-proxy-headers.conf;
+	proxy_pass http://$sync;
+	proxy_read_timeout 1h;
+}
+
+# Normal sync
+location ~ ^/_matrix/client/(api/v1|r0|v3)/events$ {
+	include snippets/matrix-proxy-headers.conf;
```
Comment (dklimpel) [inline docs/workers.md:389]: What does it include? Is there an explanation?

Comment (olmari) [reply docs/workers.md:389]: @dklimpel Oh, yes... We have all needed synapse/matrix related headers there, essentially ours looks like this:

```
proxy_set_header X-Forwarded-For $remote_addr;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header Host $host;
proxy_http_version 1.1;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection $connection_upgrade;
```

They're not even relevant to this load balancing example directly, while ofcourse such headers is _in general_ needed.. We though hard how this example for targeting certain username per worker could be best represented in workers instructions, but we didn't really figure out other than this relatively massive example that includes syncs too, as syncs themselves relates to _why_ whole ordeal is done in first place... @tswfi I think we should drop the snippet from the example as it is not directly worker related(?)

File: docs/workers.md (hunk 1 above)
Comment (tswfi) [reply docs/workers.md]: Thanks! I didn't know there is a nginx specific formatter available!

File: docs/workers.md (hunk 2 above)
Comment (tswfi) [reply docs/workers.md]: Yeah, @olmari removed those lines that are not directly related to this sticky lb routing

**Review by reivilibre — APPROVED** [review]
Thank you very much; there's quite a bit to piece together here so it's nice to leave an example in the docs!

**Pull request merged**